### PR TITLE
Sa 2315 test sd ks w prod pwsh module

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [SA-2315](https://jumpcloud.atlassian.net/browse/SA-2315) - Update sudo attribute object for JCSystem/ JCUser Associations

## What does this solve?

An update to the underlying SDK module has been preventing us from copying associations with sudo attributes to another compatible JC object. This addresses they type issue by casting the attribute as a dictionary type (Required by the SDKs).

## Is there anything particularly tricky?

No

## How should this be tested?

In JumpCloud Console, assign a user to a system with sudo attributes.
Run: 
```powershell
# this user is assigned as sudo to a system 
$userIdAssignedToSystem = '5ef5fe1613c9e332cbed14fb'
# this user has no system associations
$userIdToCopy = '5ef5fe69a899bd46963ab24a'
Copy-jcAssociation -id $userIdAssignedToSystem -TargetId $userIdToCopy -Type user -force
```
the command should fail

Now import this module from this branch:

```powershell
Import-Module "/Users/yourUsername/Documents/GitHub/support/PowerShell/JumpCloud Module/JumpCloud.psd1" -force
# this user is assigned as sudo to a system 
$userIdAssignedToSystem = '5ef5fe1613c9e332cbed14fb'
# this user has no system associations
$userIdToCopy = '5ef5fe69a899bd46963ab24a'
Copy-jcAssociation -id $userIdAssignedToSystem -TargetId $userIdToCopy -Type user -force
```
This command should work

## Screenshots
Screen Shot 2022-02-24 at 3 06 01 PM

### CODE REVIEW

File: PowerShell/Deploy/Setup-Dependencies.ps1 [hunk 1]
```diff
@@ -93,7 +93,7 @@ If (-not [System.String]::IsNullOrEmpty($Psd1))
                 # Set Module Path from required Module
                 $ModulePath = "$($LocalPSModulePath)/$($RequiredModule)"
                 # Search for module in localPSModulePath
-                $moduleFound = Get-InstalledPSResource -name $RequiredModule -path $LocalPSModulePath
```
Comment (jworkmanjc) [inline PowerShell/Deploy/Setup-Dependencies.ps1:96]: PowerShellGet 3.0.12‑beta 
Get-InstalledPSResource == Get-PSResource

File: PowerShell/JumpCloud Module/Private/Association/Invoke-JCAssociation.ps1 [hunk 2]
```diff
@@ -150,23 +150,28 @@
...
+                                                $attributeDictionary.sudo = @{'enabled' = $true; 'withoutPassword' = $true }
+                                                $Command_Template_Associations_Targets_Post = 'JumpCloud.SDK.V2\Set-JcSdk{0}Association -{0}Id:("{1}") -Id:("{2}") -Op:("{3}") -Type:("{4}") -Attributes:("{5}")'
                                             }
                                             if (-Not $additionalProperties.sudo.enabled -And -Not $additionalProperties.sudo.withoutPassword)
                                             {
                                                 $Command_Template_Associations_Targets_Post = 'JumpCloud.SDK.V2\Set-JcSdk{0}Association -{0}Id:("{1}") -Id:("{2}") -Op:("{3}") -Type:("{4}")'
                                             }
-                                            $foundAttributes
+                                            # Return Attributes
+                                            $attributeDictionary
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Private/Association/Invoke-JCAssociation.ps1:174]: Return a dotnet dictionary object when we determine what attributes are held by the first user/ system object

File: PowerShell/JumpCloud Module/Public/Commands/Invoke-JCDeployment.ps1 [hunk 3]
```diff
@@ -149,7 +149,7 @@ The CommandID will be the 24 character string populated for the _id field.')]
         $ExistingCommand = Get-JCSDKCommand -Id $CommandID
 
         # set command w/ origional type and new trigger
-        $Command = Set-JCSDKCommand -ID $CommandID -launchType trigger -trigger $trigger -Command $ExistingCommand.Command1 -CommandType $ExistingCommand.CommandType -Shell $ExistingCommand.Shell
+        $Command = Set-JCSDKCommand -ID $CommandID -launchType trigger -trigger $trigger -Command $ExistingCommand.Command1 -CommandType $ExistingCommand.CommandType -Shell $ExistingCommand.Shell -Name $ExistingCommand.Name
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Public/Commands/Invoke-JCDeployment.ps1:152]: The Invoke-JCDeployment tests had been skipped in the main branch. Function updated to include a required field in the API "name"

File: PowerShell/JumpCloud Module/Tests/Public/Utilities/Backup-JCOrganization.tests.ps1 [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Describe -Tag:('JCBackup') "Backup-JCOrganization" {
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Tests/Public/Utilities/Backup-JCOrganization.tests.ps1:1]: JCBackupOrg should be different from JCBackup, the backup-jcorganization function technically is a different function and makes use of our SDKs whereas the original backup function does not

**Review by kmaranionjc — APPROVED** [review]
Tests are good. Nice work!

**Pull request merged**